platform/core/src/services/MultiMonitorService.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * This service manages multiple monitors or windows.
+ */
+export class MultiMonitorService {
+  public readonly numberOfScreens: number;
+  private windowsConfig;
+  private screenConfig;
+  private launchWindows = [];
+  private commandsManager;
+
+  public readonly screenNumber: number;
+  public readonly isMultimonitor: boolean;
+
+  public static REGISTRATION = {
+    name: 'multiMonitorService',
+    create: ({ configuration, commandsManager }): MultiMonitorService => {
+      const service = new MultiMonitorService(configuration, commandsManager);
+      return service;
+    },
+  };
+
+  constructor(configuration, commandsManager) {
+    const params = new URLSearchParams(window.location.search);
+    const screenNumber = params.get('screenNumber');
+    const multimonitor = params.get('multimonitor');
+    const testParams = { params, screenNumber, multimonitor };
+    this.screenNumber = screenNumber ? Number(screenNumber) : 0;
+    this.commandsManager = commandsManager;
+    const windowAny = window as any;
+    windowAny.multimonitor ||= {
+      setLaunchWindows: this.setLaunchWindows,
+      launchWindows: this.launchWindows,
+      commandsManager,
+    };
+    windowAny.multimonitor.commandsManager = commandsManager;
+    this.launchWindows = (window as any).multimonitor?.launchWindows || this.launchWindows;
+    if (!this.screenNumber) {
+      this.launchWindows[0] = window;
+    }
+    windowAny.commandsManager = (...args) => configuration.commandsManager;
+    for (const windowsConfig of Array.isArray(configuration) ? configuration : []) {
+      if (windowsConfig.test(testParams)) {
+        this.isMultimonitor = true;
+        this.numberOfScreens = windowsConfig.screens.length;
+        this.windowsConfig = windowsConfig;
+        this.screenConfig = windowsConfig.screens[this.screenNumber];
+        if (!this.screenConfig) {
+          throw new Error(`Screen ${screenNumber} not configured in ${this.windowsConfig}`);
+        }
+        window.name = this.screenConfig.id;
+        return;
+      }
+    }
+    this.numberOfScreens = 1;
+    this.isMultimonitor = false;
+  }
+
+  public run(screenDelta = 1, commands, options) {
+    const screenNumber = (this.screenNumber + (screenDelta ?? 1)) % this.numberOfScreens;
```

Comment:
Using negative delta to run commands from window 2 into window 1 feels a bit weird I think. I could see explicit screen id being more intuitive. Or what's the use case with delta